external/fv3fit/fv3fit/testing.py

```diff
@@ -67,13 +73,14 @@ def predict(self, X: xr.Dataset) -> xr.Dataset:
             else:
                 array = np.full([n_samples], float(output))
                 data_vars[name] = xr.DataArray(data=array, dims=[self.sample_dim_name])
-        if self.sample_dim_name in X.coords:
-            coords: Optional[Mapping[Hashable, Any]] = {
-                self.sample_dim_name: X.coords[self.sample_dim_name]
-            }
-        else:
-            coords = None
-        return xr.Dataset(data_vars=data_vars, coords=coords)
+        coords: Optional[Mapping[Hashable, Any]] = {
+            self.sample_dim_name: stacked_X.coords[self.sample_dim_name]
+        }
```

Comment:
There's nothing here to guarantee that self.sample_dim_name is in stacked_X.coords, but this should be fixed if you use a global SAMPLE_DIM_NAME here that is also used by `stack_non_vertical`.